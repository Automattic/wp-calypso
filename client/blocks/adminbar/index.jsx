/**
 * External dependencies
 */
import React, { Component } from 'react';

/**
 * Internal dependencies
 */
import Button from 'components/button';
import Card from 'components/card';
import Gridicon from 'components/gridicon';
import Popover from 'components/popover';

class AdminBar extends Component {

	state = {
		isWpMenuVisible: false
	};

	toggleWpMenu = ( isWpMenuVisible ) => this.setState( { isWpMenuVisible } );

	hideWpMenu = () => this.toggleWpMenu( false );
	showWpMenu = () => this.toggleWpMenu( true );

	setPopoverContext = ( popoverContext ) => {
		if ( popoverContext ) {
			this.setState( { popoverContext } );
		}
	};

	render() {
		const { isWpMenuVisible, popoverContext } = this.state;

		return (
			<Card compact className="adminbar">
				<Button borderless>
					<Gridicon icon="ellipsis" />
				</Button>
				<Button
					borderless
					ref={ this.setPopoverContext }
					onClick={ isWpMenuVisible ? this.hideWpMenu : this.showWpMenu }>
					<Gridicon icon="my-sites" />
				</Button>
				<Popover
					className="adminbar__main-menu"
					onClose={ this.hideWpMenu }
					isVisible={ isWpMenuVisible }
					context={ popoverContext }
					position="top left"
				>
					<Button borderless>
						<Gridicon icon="my-sites" /> My Sites
					</Button>
					<Button borderless>
						<Gridicon icon="reader" /> Reader
					</Button>
					<Button borderless>
						<Gridicon icon="user-circle" /> Me
					</Button>
					<Button borderless>
						<Gridicon icon="bell" /> Notifications
					</Button>
				</Popover>
			</Card>
		);
	}
}

export default AdminBar;
