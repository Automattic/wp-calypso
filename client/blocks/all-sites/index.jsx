/**
 * External dependencies
 */

import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { localize } from 'i18n-calypso';
import { noop } from 'lodash';
import classNames from 'classnames';

/**
 * Internal dependencies
 */
import AllSitesIcon from './all-sites-icon';
import config from 'calypso/config';
import Count from 'calypso/components/count';
import getSites from 'calypso/state/selectors/get-sites';
import { getCurrentUserVisibleSiteCount } from 'calypso/state/current-user/selectors';

/**
 * Style dependencies
 */
import './style.scss';

class AllSites extends Component {
	static defaultProps = {
		onSelect: noop,
		href: null,
		isSelected: false,
		isHighlighted: false,
		showCount: true,
		domain: '',
	};

	static propTypes = {
		onSelect: PropTypes.func,
		href: PropTypes.string,
		isSelected: PropTypes.bool,
		isHighlighted: PropTypes.bool,
		showCount: PropTypes.bool,
		count: PropTypes.number,
		title: PropTypes.string,
		domain: PropTypes.string,
		onMouseEnter: PropTypes.func,
		onMouseLeave: PropTypes.func,
	};

	onSelect = ( event ) => {
		this.props.onSelect( event );
	};

	renderSiteCount() {
		return <Count count={ this.props.count } />;
	}

	render() {
		const {
			title,
			href,
			domain,
			sites,
			translate,
			isHighlighted,
			isSelected,
			showCount,
		} = this.props;

		// Note: Update CSS selectors in SiteSelector.scrollToHighlightedSite() if the class names change.
		const allSitesClass = classNames( {
			'all-sites': true,
			'is-selected': isSelected,
			'is-highlighted': isHighlighted,
		} );

		return (
			<div className={ allSitesClass }>
				<a
					className="all-sites__content site__content"
					href={ href }
					onMouseEnter={ this.props.onMouseEnter }
					onMouseLeave={ this.props.onMouseLeave }
					onClick={ this.onSelect }
				>
					{ showCount && this.renderSiteCount() }
					<div className="all-sites__info site__info">
						<span className="all-sites__title site__title">
							{ title || translate( 'All My Sites' ) }
						</span>
						{ domain && <span className="all-sites__domain site__domain">{ domain }</span> }
						<AllSitesIcon sites={ sites } />
					</div>
				</a>
			</div>
		);
	}
}

// don't instantiate function in `connect`
const isSiteVisible = ( { visible = true } ) => visible;

export default connect( ( state, props ) => {
	// If sites or count are not specified as props, fetch the default values from Redux
	const {
		sites = getSites( state ),
		userSitesCount = getCurrentUserVisibleSiteCount( state ),
	} = props;

	const visibleSites = sites?.filter( isSiteVisible );

	return {
		sites: config.isEnabled( 'realtime-site-count' ) ? visibleSites : sites,
		count: config.isEnabled( 'realtime-site-count' ) ? visibleSites.length : userSitesCount,
	};
} )( localize( AllSites ) );
