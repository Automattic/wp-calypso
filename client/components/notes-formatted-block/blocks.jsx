/**
 * External dependencies
 */
import React from 'react';
import { startsWith } from 'lodash';

/**
 * Internal dependencies
 */
import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';

const blocksByType = {};
export function getBlockByType( type ) {
	return blocksByType[ type ];
}

function assignBlockType( type, block ) {
	blocksByType[ type ] = block;
}

export const Strong = ( { children } ) => <strong>{ children }</strong>;
assignBlockType( 'b', Strong );

export const Emphasis = ( { children } ) => <em>{ children }</em>;
assignBlockType( 'i', Emphasis );

export const Preformatted = ( { children } ) => <pre>{ children }</pre>;
assignBlockType( 'pre', Preformatted );

export const Link = ( { content, onClick, children } ) => {
	const { url: originalUrl, activity, section, intent } = content;

	const isWordPressLink = startsWith( originalUrl, 'https://wordpress.com' );

	// Don't render links to WordPress.com inside Jetpack Cloud
	if ( isWordPressLink && isJetpackCloud() ) {
		return children;
	}

	// On Calypso, relativize links to WordPress.com;
	// for other destinations, render a link with the URL as-is
	const linkUrl = isWordPressLink ? originalUrl.substr( 21 ) : originalUrl;
	return (
		<a
			href={ linkUrl }
			onClick={ onClick }
			data-activity={ activity }
			data-section={ section }
			data-intent={ intent }
		>
			{ children }
		</a>
	);
};
assignBlockType( 'a', Link );
assignBlockType( 'link', Link );

export const FilePath = ( { children } ) => (
	<div>
		<code>{ children }</code>
	</div>
);
assignBlockType( 'filepath', FilePath );

export const Post = ( { content, children } ) => {
	// Don't render links to WordPress.com inside Jetpack Cloud
	if ( isJetpackCloud() ) {
		return content.isTrashed ? children : <em>{ children }</em>;
	}

	if ( content.isTrashed ) {
		return <a href={ `/posts/${ content.siteId }/trash` }>{ children }</a>;
	}

	return (
		<a href={ `/read/blogs/${ content.siteId }/posts/${ content.postId }` }>
			<em>{ children }</em>
		</a>
	);
};
assignBlockType( 'post', Post );

export const Comment = ( { content, children } ) => {
	// Don't render links to WordPress.com inside Jetpack Cloud
	if ( isJetpackCloud() ) {
		return children;
	}

	return (
		<a
			href={ `/read/blogs/${ content.siteId }/posts/${ content.postId }#comment-${ content.commentId }` }
		>
			{ children }
		</a>
	);
};
assignBlockType( 'comment', Comment );

export const Person = ( { content, onClick, meta, children } ) => {
	// Don't render links to WordPress.com inside Jetpack Cloud
	if ( isJetpackCloud() ) {
		return <strong>{ children }</strong>;
	}

	return (
		<a
			href={ `/people/edit/${ content.siteId }/${ content.name }` }
			onClick={ onClick }
			data-activity={ meta.activity }
			data-section="users"
			data-intent="edit"
		>
			<strong>{ children }</strong>
		</a>
	);
};
assignBlockType( 'person', Person );

export const Plugin = ( { content, onClick, meta, children } ) => {
	// Don't render links to WordPress.com inside Jetpack Cloud
	if ( isJetpackCloud() ) {
		return children;
	}

	return (
		<a
			href={ `/plugins/${ content.pluginSlug }/${ content.siteSlug }` }
			onClick={ onClick }
			data-activity={ meta.activity }
			data-section="plugins"
			data-intent="view"
		>
			{ children }
		</a>
	);
};
assignBlockType( 'plugin', Plugin );

export const Theme = ( { content, onClick, meta, children } ) => {
	const { themeUri, themeSlug, siteSlug } = content;
	if ( ! themeUri ) {
		return children;
	}

	if ( /wordpress\.com/.test( themeUri ) ) {
		// Don't render links to WordPress.com inside Jetpack Cloud
		return isJetpackCloud() ? (
			children
		) : (
			<a
				href={ `/theme/${ themeSlug }/${ siteSlug }` }
				onClick={ onClick }
				data-activity={ meta.activity }
				data-section="themes"
				data-intent="view"
			>
				{ children }
			</a>
		);
	}

	return (
		<a
			href={ themeUri }
			target="_blank"
			rel="noopener noreferrer"
			onClick={ onClick }
			data-activity={ meta.activity }
			data-section="themes"
			data-intent="view"
		>
			{ children }
		</a>
	);
};
assignBlockType( 'theme', Theme );
