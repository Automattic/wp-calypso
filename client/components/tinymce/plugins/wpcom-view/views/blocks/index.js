/**
 * External dependencies
 */
import React from 'react';
import debugFactory from 'debug';

/**
 * Internal dependencies
 */
//import { parse } from './post-parser';

const debug = debugFactory( 'block-editor' );

const blockOpen = /<!-- wp:([a-z-]+) ([a-z: ]*) ?-->/g;
const blockClose = /<!-- \/wp -->/g;

export function match( content ) {
	blockOpen.lastIndex = 0;
	const openMatch = blockOpen.exec( content );

	if ( ! openMatch ) {
		debug( 'no openMatch', content );
		return;
	}

	blockClose.lastIndex = openMatch.index;
	debug( 'setting index', openMatch.index );
	const closeMatch = blockClose.exec( content );

	if ( ! closeMatch ) {
		debug( 'no closeMatch' );
		return;
	}

	const result = {
		index: openMatch.index,
		content: content.slice( openMatch.index, closeMatch.lastIndex ),
		text: content.slice(
				openMatch.index + openMatch[ 0Â ].length,
				closeMatch.index ),
	};

	debug( 'result', result );
	return result;
}

export function serialize( content ) {
	return encodeURIComponent( content );
}

export function getComponent() {
	return View;
}

export function edit( editor, content ) {
	debug( 'edit' );
	editor.execCommand( 'wpcomContactForm', content );
}

function trimBreaks( text ) {
	return text
		.trim()
		.replace( /^<br \/>/, '' )
		.replace( /<br \/>$/, '' );
}

const blockStyle = {
	margin: '0.2em',
	padding: '0.3em',
	border: '1px dashed black',
};

function View( props ) {
	const { text } = match( props.content );
	const trimmed = trimBreaks( text );

	debug( 'View', props, trimmed );

	return <div style={ blockStyle }>{ trimmed }</div>;
}
