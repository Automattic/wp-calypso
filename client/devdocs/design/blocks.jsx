/**
 * External dependencies
 */
import React from 'react';
import page from 'page';
import classnames from 'classnames';
import { trim } from 'lodash';
import { slugToCamelCase } from 'calypso/devdocs/docs-example/util';

/**
 * Internal dependencies
 */
import Collection from 'calypso/devdocs/design/search-collection';
import DocumentHead from 'calypso/components/data/document-head';
import HeaderCake from 'calypso/components/header-cake';
import Main from 'calypso/components/main';
import ReadmeViewer from 'calypso/components/readme-viewer';
import SearchCard from 'calypso/components/search-card';
import { isEnabled } from 'calypso/config';

/**
 * Docs examples
 */
import AllSites from 'calypso/blocks/all-sites/docs/example';
import CreditCardForm from 'calypso/blocks/credit-card-form/docs/example';
import CalendarButton from 'calypso/blocks/calendar-button/docs/example';
import CalendarPopover from 'calypso/blocks/calendar-popover/docs/example';
import AuthorSelector from 'calypso/blocks/author-selector/docs/example';
import CommentButtons from 'calypso/blocks/comment-button/docs/example';
import FollowButton from 'calypso/blocks/follow-button/docs/example';
import LikeButtons from 'calypso/blocks/like-button/docs/example';
import PostSchedule from 'calypso/components/post-schedule/docs/example';
import PostSelector from 'calypso/my-sites/post-selector/docs/example';
import ProductPlanOverlapNotices from 'calypso/blocks/product-plan-overlap-notices/docs/example';
import ProductSelector from 'calypso/blocks/product-selector/docs/example';
import Site from 'calypso/blocks/site/docs/example';
import SitePlaceholder from 'calypso/blocks/site/docs/placeholder-example';
import SitesDropdown from 'calypso/components/sites-dropdown/docs/example';
import SiteIcon from 'calypso/blocks/site-icon/docs/example';
import Theme from 'calypso/components/theme/docs/example';
import HappinessSupport from 'calypso/components/happiness-support/docs/example';
import ThemesListExample from 'calypso/components/themes-list/docs/example';
import PlanStorage from 'calypso/blocks/plan-storage/docs/example';
import PlanCompareCard from 'calypso/my-sites/plan-compare-card/docs/example';
import DomainTip from 'calypso/blocks/domain-tip/docs/example';
import PostItem from 'calypso/blocks/post-item/docs/example';
import PostStatus from 'calypso/blocks/post-status/docs/example';
import ReaderAuthorLink from 'calypso/blocks/reader-author-link/docs/example';
import ReaderSiteStreamLink from 'calypso/blocks/reader-site-stream-link/docs/example';
import AuthorCompactProfile from 'calypso/blocks/author-compact-profile/docs/example';
import RelatedPostCard from 'calypso/blocks/reader-related-card/docs/example';
import PlanPrice from 'calypso/my-sites/plan-price/docs/example';
import PostShare from 'calypso/blocks/post-share/docs/example';
import PlanThankYouCard from 'calypso/blocks/plan-thank-you-card/docs/example';
import DismissibleCard from 'calypso/blocks/dismissible-card/docs/example';
import PostEditButton from 'calypso/blocks/post-edit-button/docs/example';
import PostComment from 'calypso/blocks/comments/docs/post-comment-example';
import ReaderAvatar from 'calypso/blocks/reader-avatar/docs/example';
import SubscriptionLengthPicker from 'calypso/blocks/subscription-length-picker/docs/example';
import ImageEditor from 'calypso/blocks/image-editor/docs/example';
import ImageSelector from 'calypso/blocks/image-selector/docs/example';
import VideoEditor from 'calypso/blocks/video-editor/docs/example';
import ReaderPostCard from 'calypso/blocks/reader-post-card/docs/example';
import ReaderCombinedCard from 'calypso/blocks/reader-combined-card/docs/example';
import ReaderRecommendedSites from 'calypso/blocks/reader-recommended-sites/docs/example';
import ReaderPostOptionsMenu from 'calypso/blocks/reader-post-options-menu/docs/example';
import DailyPostButton from 'calypso/blocks/daily-post-button/docs/example';
import ReaderSubscriptionListItem from 'calypso/blocks/reader-subscription-list-item/docs/example';
import PostLikes from 'calypso/blocks/post-likes/docs/example';
import ReaderFeaturedVideo from 'calypso/blocks/reader-featured-video/docs/example';
import NpsSurvey from 'calypso/blocks/nps-survey/docs/example';
import ReaderExportButton from 'calypso/blocks/reader-export-button/docs/example';
import ReaderImportButton from 'calypso/blocks/reader-import-button/docs/example';
import SharingPreviewPane from 'calypso/blocks/sharing-preview-pane/docs/example';
import ReaderShare from 'calypso/blocks/reader-share/docs/example';
import Login from 'calypso/blocks/login/docs/example';
import LocationSearch from 'calypso/blocks/location-search/docs/example';
import ConversationCommentList from 'calypso/blocks/conversations/docs/example';
import SimplePaymentsDialog from 'calypso/components/tinymce/plugins/simple-payments/dialog/docs/example';
import ConversationCaterpillar from 'calypso/blocks/conversation-caterpillar/docs/example';
import ConversationFollowButton from 'calypso/blocks/conversation-follow-button/docs/example';
import ColorSchemePicker from 'calypso/blocks/color-scheme-picker/docs/example';
import UserMentions from 'calypso/blocks/user-mentions/docs/example';
import SupportArticleDialog from 'calypso/blocks/support-article-dialog/docs/example';
import TimeMismatchWarning from 'calypso/blocks/time-mismatch-warning/docs/example';
import UpsellNudge from 'calypso/blocks/upsell-nudge/docs/example';

export default class AppComponents extends React.Component {
	static displayName = 'AppComponents';
	state = { filter: '' };

	onSearch = ( term ) => {
		this.setState( { filter: trim( term || '' ).toLowerCase() } );
	};

	backToComponents = () => {
		page( '/devdocs/blocks/' );
	};

	render() {
		const className = classnames( 'devdocs', 'devdocs__blocks', {
			'is-single': this.props.component,
			'is-list': ! this.props.component,
		} );

		return (
			<Main className={ className }>
				<DocumentHead title="Blocks" />

				{ this.props.component ? (
					<React.Fragment>
						<HeaderCake onClick={ this.backToComponents } backText="All Blocks">
							{ slugToCamelCase( this.props.component ) }
						</HeaderCake>
						{ isEnabled( 'devdocs/color-scheme-picker' ) && (
							<ColorSchemePicker readmeFilePath="color-scheme-picker" />
						) }
					</React.Fragment>
				) : (
					<div>
						<ReadmeViewer readmeFilePath="/client/devdocs/blocks/README.md" />
						<SearchCard
							onSearch={ this.onSearch }
							initialValue={ this.state.filter }
							placeholder="Search blocksâ€¦"
							analyticsGroup="Docs"
						/>
					</div>
				) }
				<Collection
					component={ this.props.component }
					filter={ this.state.filter }
					section="blocks"
				>
					{ isEnabled( 'devdocs/color-scheme-picker' ) && (
						<ColorSchemePicker readmeFilePath="color-scheme-picker" />
					) }
					<AllSites readmeFilePath="all-sites" />
					<AuthorSelector readmeFilePath="author-selector" />
					<CalendarButton readmeFilePath="calendar-button" />
					<CalendarPopover readmeFilePath="calendar-popover" />
					<CommentButtons readmeFilePath="comment-button" />
					<CreditCardForm readmeFilePath="credit-card-form" />
					<FollowButton readmeFilePath="follow-button" />
					<HappinessSupport />
					<ImageEditor readmeFilePath="image-editor" />
					<VideoEditor readmeFilePath="video-editor" />
					<LikeButtons readmeFilePath="like-button" />
					<Login />
					<LocationSearch readmeFilePath="location-search" />
					<PostEditButton />
					<PlanStorage readmeFilePath="plan-storage" />
					<PostSchedule />
					<PostSelector />
					<ProductPlanOverlapNotices readmeFilePath="product-plan-overlap-notices" />
					<ProductSelector readmeFilePath="product-selector" />
					<Site readmeFilePath="site" />
					<SitePlaceholder />
					<SitesDropdown />
					<SiteIcon readmeFilePath="site-icon" />
					<Theme />
					<ThemesListExample />
					<PlanCompareCard />
					<DomainTip />
					<RelatedPostCard />
					<PostItem readmeFilePath="post-item" />
					<PostStatus readmeFilePath="post-status" />
					<ReaderAuthorLink readmeFilePath="reader-author-link" />
					<ReaderSubscriptionListItem />
					<ReaderSiteStreamLink readmeFilePath="reader-site-stream-link" />
					<AuthorCompactProfile />
					<ReaderPostCard />
					<ReaderCombinedCard />
					<ReaderRecommendedSites />
					<PlanPrice />
					<PostShare readmeFilePath="post-share" />
					<PlanThankYouCard readmeFilePath="plan-thank-you-card" />
					<DismissibleCard readmeFilePath="dismissible-card" />
					<ReaderAvatar readmeFilePath="reader-avatar" />
					<ReaderPostOptionsMenu readmeFilePath="reader-post-options-menu" />
					<DailyPostButton readmeFilePath="daily-post-button" />
					<PostLikes readmeFilePath="post-likes" />
					<ReaderFeaturedVideo readmeFilePath="reader-featured-video" />
					{ isEnabled( 'nps-survey/devdocs' ) && <NpsSurvey readmeFilePath="nps-survey" /> }
					<ReaderExportButton readmeFilePath="reader-export-button" />
					<ReaderImportButton readmeFilePath="reader-import-button" />
					<SharingPreviewPane />
					<SimplePaymentsDialog />
					<SubscriptionLengthPicker />
					<ReaderShare readmeFilePath="reader-share" />
					<ConversationCommentList />
					<PostComment />
					<ConversationCaterpillar readmeFilePath="conversation-caterpillar" />
					<ConversationFollowButton />
					{ isEnabled( 'reader/user-mention-suggestions' ) && (
						<UserMentions readmeFilePath="user-mentions" />
					) }
					<SupportArticleDialog />
					<ImageSelector readmeFilePath="image-selector" />
					<TimeMismatchWarning readmeFilePath="time-mismatch-warning" />
					<UpsellNudge />
				</Collection>
			</Main>
		);
	}
}
