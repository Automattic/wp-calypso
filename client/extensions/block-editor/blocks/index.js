/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import CodeBlock from './code';
import GenericBlock from './generic';

export function renderBlock( block ) {
	if ( block.type === 'Text' ) {
		return <p>{ block.value }</p>;
	}

	if ( block.blockType === 'code' ) {
		return <CodeBlock { ...block } />;
	}

	if ( block.type === 'WP_Block' ) {
		return <GenericBlock { ...block } />;
	}

	return null;
}
