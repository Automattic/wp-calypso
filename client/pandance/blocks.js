/**
 * External dependencies
 */
import React from 'react';
import page from 'page';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

/***
 * Internal Dependencies
 */
import Card from 'components/card';
import Ribbon from 'components/ribbon';
import Button from 'components/button';
import { toggleBlock } from 'state/pandance/actions';
import SITE_BLOCKS from './site-blocks';

class BlockItem extends React.Component {

	handleClick = () => {
		this.props.onBlockToggle( this.props.id );
	};

	render() {
		return <Card onClick={ this.handleClick }>
			{ this.props.isSelected && <Ribbon>Selected</Ribbon> }
			<p>{ this.props.name }</p>
		</Card>;
	}
}

export class Blocks extends React.Component {
	constructor( props ) {
		super( props );

		this.boundOnBlockToggle = this.onBlockToggle.bind( this );
	}

	onBlockToggle( id ) {
		this.props.toggleBlock( id );
	}

	render() {
		return <div className="options wrapper">
			<h2>Choose a few things to get your site started.</h2>
			<p>We will help you start a site so you can tell the world about your business right away. This is just the start of your online presence, you can add to it later.</p>
				{
					SITE_BLOCKS.map( block => <BlockItem
						id={ block.id }
						name={ block.name }
						isSelected={ this.props.selected.includes( block.id ) }
						onBlockToggle={ this.boundOnBlockToggle } />
					)
				}
			<div className="button-container">
				<Button primary={ true } onClick={ () => page( '/pandance/info' ) }>Continue</Button>
			</div>
		</div>;
	}
}

export default connect( ( state, props ) => ( {
	selected: state.pandance.selected,
} ), dispatch => bindActionCreators( { toggleBlock }, dispatch ) )( Blocks );
