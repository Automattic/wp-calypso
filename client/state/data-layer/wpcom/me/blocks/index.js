/*
 * @format
 */

/**
 * Internal dependencies
 */
import { mergeHandlers } from 'state/action-watchers/utils';
import { registerHandlers } from 'state/data-layer/handler-registry';
import sites from './sites';

registerHandlers( 'state/data-layer/wpcom/me/blocks/index.js', mergeHandlers( sites ) );

export default {};
