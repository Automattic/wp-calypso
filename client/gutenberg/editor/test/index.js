/**
 * @format
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import { noop } from 'lodash';

/**
 * Internal dependencies
 */
import {
	registerStore,
	registerReducer,
	registerSelectors,
	dispatch,
	select,
} from '@wordpress/data';

import { createBlock, registerBlockType } from '@wordpress/blocks';

import { parse } from '@wordpress/block-serialization-spec-parser';

import apiFetch from '@wordpress/api-fetch';

describe( 'Gutenberg @wordress/data package', () => {
	describe( 'registerStore', () => {
		it( 'should be shorthand for reducer, actions, selectors registration', () => {
			const DEFAULT_STATE = {
				prices: { shirt: 5 },
				discountPercent: 0,
			};

			const store = registerStore( 'my-shop', {
				reducer( state = DEFAULT_STATE, action ) {
					switch ( action.type ) {
						case 'SET_PRICE':
							return {
								...state,
								prices: {
									...state.prices,
									[ action.item ]: action.price,
								},
							};

						case 'START_SALE':
							return {
								...state,
								discountPercent: action.discountPercent,
							};
					}

					return state;
				},

				actions: {
					setPrice( item, price ) {
						return {
							type: 'SET_PRICE',
							item,
							price,
						};
					},
					startSale( discountPercent ) {
						return {
							type: 'START_SALE',
							discountPercent,
						};
					},
				},

				selectors: {
					getPrice( state, item ) {
						const { prices, discountPercent } = state;
						const price = prices[ item ];

						return price * ( 1 - 0.01 * discountPercent );
					},
				},
			} );

			expect( store.getState() ).toEqual( DEFAULT_STATE );
			expect( select( 'my-shop' ) ).toHaveProperty( 'getPrice' );
			expect( select( 'my-shop' ).getPrice( 'shirt' ) ).toEqual( 5 );
			dispatch( 'my-shop' ).setPrice( 'shirt', 10 );
			expect( select( 'my-shop' ).getPrice( 'shirt' ) ).toEqual( 10 );
			dispatch( 'my-shop' ).startSale( 20 );
			expect( select( 'my-shop' ).getPrice( 'shirt' ) ).toEqual( 8 );
		} );
	} );

	describe( 'registerReducer', () => {
		it( 'Should append reducers to the state', () => {
			const reducer = () => 'tea';

			const store = registerReducer( 'reducer', reducer );
			expect( store.getState() ).toEqual( 'tea' );
		} );
	} );

	describe( 'select', () => {
		it( 'registers multiple selectors to the public API', () => {
			const store = registerReducer( 'reducer', () => {} );
			const selector = jest.fn( () => 'test-result' );

			registerSelectors( 'reducer', { selector } );

			expect( select( 'reducer' ).selector() ).toEqual( 'test-result' );
			expect( selector ).toBeCalledWith( store.getState() );
		} );
	} );
} );

describe( 'Gutenberg @wordress/blocks package', () => {
	describe( 'createBlock', () => {
		it( 'should create a block given its blockType, attributes, inner blocks', () => {
			registerBlockType( 'core/test-block', {
				attributes: {
					align: {
						type: 'string',
					},
					includesDefault: {
						type: 'boolean',
						default: true,
					},
					includesFalseyDefault: {
						type: 'number',
						default: 0,
					},
				},
				save: noop,
				category: 'common',
				title: 'test block',
			} );

			const block = createBlock( 'core/test-block', { align: 'left' }, [
				createBlock( 'core/test-block' ),
			] );

			expect( block.name ).toEqual( 'core/test-block' );
			expect( block.attributes ).toEqual( {
				includesDefault: true,
				includesFalseyDefault: 0,
				align: 'left',
			} );
			expect( block.isValid ).toBe( true );
			expect( block.innerBlocks ).toHaveLength( 1 );
			expect( block.innerBlocks[ 0 ].name ).toBe( 'core/test-block' );
			expect( typeof block.clientId ).toBe( 'string' );
		} );
	} );
} );

describe( 'block-serialization-spec-parser', () => {
	// Try test from https://github.com/WordPress/gutenberg/blob/54990c0f77294e6603cc4e0aa957c0679c1bf242/packages/block-serialization-spec-parser/test/index.js
	test( 'parse() works properly', () => {
		const result = parse( '<!-- wp:core/more --><!--more--><!-- /wp:core/more -->' );

		expect( result ).toMatchSnapshot();
	} );
} );

describe( 'Gutenberg @wordress/api-fetch package', () => {
	// Try tests from https://github.com/WordPress/gutenberg/blob/54990c0f77294e6603cc4e0aa957c0679c1bf242/packages/api-fetch/src/test/index.js
	describe( 'apiFetch', () => {
		const originalFetch = window.fetch;
		beforeAll( () => {
			window.fetch = jest.fn();
		} );

		afterAll( () => {
			window.fetch = originalFetch;
		} );

		it( 'should call the API propertly', () => {
			window.fetch.mockReturnValue(
				Promise.resolve( {
					status: 200,
					json() {
						return Promise.resolve( { message: 'ok' } );
					},
				} )
			);

			return apiFetch( { path: '/random' } ).then( body => {
				expect( body ).toEqual( { message: 'ok' } );
			} );
		} );

		it( 'should return the error message properly', () => {
			window.fetch.mockReturnValue(
				Promise.resolve( {
					status: 400,
					json() {
						return Promise.resolve( {
							code: 'bad_request',
							message: 'Bad Request',
						} );
					},
					clone() {
						return this;
					},
				} )
			);

			return apiFetch( { path: '/random' } ).catch( body => {
				expect( body ).toEqual( {
					code: 'bad_request',
					message: 'Bad Request',
				} );
			} );
		} );

		it( 'should return invalid JSON error if no json response', () => {
			window.fetch.mockReturnValue(
				Promise.resolve( {
					status: 200,
				} )
			);

			return apiFetch( { path: '/random' } ).catch( body => {
				expect( body ).toEqual( {
					code: 'invalid_json',
					message: 'The response is not a valid JSON response.',
				} );
			} );
		} );

		it( 'should return invalid JSON error if response is not valid', () => {
			window.fetch.mockReturnValue(
				Promise.resolve( {
					status: 200,
					json() {
						return Promise.reject();
					},
				} )
			);

			return apiFetch( { path: '/random' } ).catch( body => {
				expect( body ).toEqual( {
					code: 'invalid_json',
					message: 'The response is not a valid JSON response.',
				} );
			} );
		} );

		it( 'should not try to parse the response', () => {
			window.fetch.mockReturnValue(
				Promise.resolve( {
					status: 200,
				} )
			);

			return apiFetch( { path: '/random', parse: false } ).then( response => {
				expect( response ).toEqual( {
					status: 200,
				} );
			} );
		} );

		it( 'should not try to parse the error', () => {
			window.fetch.mockReturnValue(
				Promise.resolve( {
					status: 400,
				} )
			);

			return apiFetch( { path: '/random', parse: false } ).catch( response => {
				expect( response ).toEqual( {
					status: 400,
				} );
			} );
		} );
	} );
} );
