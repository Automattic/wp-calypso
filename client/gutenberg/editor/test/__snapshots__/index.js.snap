// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block-serialization-spec-parser parse() works properly 1`] = `
Array [
  Object {
    "attrs": Object {},
    "blockName": "core/more",
    "innerBlocks": Array [],
    "innerHTML": "<!--more-->",
  },
]
`;
